### PR TITLE
Fixed incorrect type assumptions

### PR DESCRIPTION
**Pull request by ijlee2**
## Description

I tested `v0.4.0` on a production project and found a few edge cases that will cause a runtime error. After patching the code, I checked that the codemod runs without an error. I'll work on adding tests later.

### CODE REVIEW

File: src/steps/analyze-project/analyze-components.ts [hunk 1]
```diff
@@ -31,15 +31,25 @@ export function analyzeComponents(
...
+        Element,
+      });
+    } catch (error) {
+      let message = `WARNING: analyzeComponents could not parse \`${filePath}\`. Please update the file manually.`;
+
+      if (error instanceof Error) {
+        message += ` (${error.message})`;
+      }
+
+      console.warn(`${message}\n`);
```
Comment (ijlee2) [inline src/steps/analyze-project/analyze-components.ts:51]: To help end-developers file a bug report.

File: src/steps/analyze-project/analyze-components/find-arguments.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ function analyzeClass(file?: string): Set<string> {
       const { id: leftHandSide, init: rightHandSide } = node.value;
       let isValid = false;
 
-      switch (rightHandSide.type) {
+      switch (rightHandSide?.type) {
```
Comment (ijlee2) [inline src/steps/analyze-project/analyze-components/find-arguments.ts:48]: `rightHandSide` is equal to `null` for the following code:

```ts
for (const element of someArray) {
  /* ... */
}
```

File: src/steps/create-registries/rename-component/pass-component-name-to-base-component.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ export function passComponentNameToBaseComponent(
         return false;
       }
 
-      switch (declaration.init!.type) {
+      switch (declaration.init?.type) {
```
Comment (ijlee2) [inline src/steps/create-registries/rename-component/pass-component-name-to-base-component.ts:55]: `declaration.init` is equal to `null` for the following code:

```ts
function findInteractiveElement(
  event: MouseEvent,
  parent?: HTMLElement,
): HTMLElement | undefined {
  const path = event.composedPath() as HTMLElement[];

  for (const element of path) {
    if (isInteractiveElement(element)) {
      return element;
    }

    if (element === parent) {
      return;
    }
  }

  // eslint-disable-next-line no-useless-return
  return;
}
```

File: src/steps/update-signatures/update-signature.ts [hunk 4]
```diff
@@ -43,7 +43,9 @@ export function updateSignature(file: string, data: Data): string {
       const argsNode = getBodyNode(path.node, 'Args');
 
       const isArgsKnown =
-        argsNode?.typeAnnotation.typeAnnotation.members.length > 0;
+        argsNode &&
+        argsNode.typeAnnotation.typeAnnotation.type === 'TSTypeLiteral' &&
```
Comment (ijlee2) [inline src/steps/update-signatures/update-signature.ts:47]: `argsNode.typeAnnotation.typeAnnotation.type` is not equal to `TSTypeLiteral` for the following code:

```ts 
export interface Signature {
  Args: unknown;
  Element: HTMLDivElement;
}
```

**Comment by ijlee2** [comment]
**TODO**

Ensure that the codemod can recognize an argument in a "default-value" expression such as,

```ts
const increment = this.args.increment ?? Increment.HalfHourly;
```
